ibr/ibr-sdk/build/ibr-sdk.js

```diff
@@ -22,21 +25,196 @@ typeof define === 'function' && define.amd ? define(['exports'], factory) :
+    controls.target.set( 0, 0.5, 0 );
+    controls.update();
+    controls.enablePan = false;
+    controls.enableDamping = true;
+    camera.position.set( 0, 0, 7000 );
+    camera.lookAt( 0, 0, 0 );
+    animate();
+
+    /**
+    * Renders the scene.
+    */
+    function animate() {
+      requestAnimationFrame( animate );
+      controls.update();
+      renderer.render( scene, camera );
+    }
+
+    return scene;
+  }
+
+  /**
+   * Create a new HTML Label Tag based on given name string and attach it to
+   * given parent tag.
+   * @param {string} tagName Type of tag to be created.
+   * @param {string} name InnerHTML of the Label tag.
+   * @param {tag} parentTag Tag to be attached to by the newly created label
+    tag.
+   * @param {string} [forId=undefined] ID to be set as the value of the newly
+    created label tag's for attribute. Needed if Label tag is created for a
+    checkbox.
+   * @return {tag} Newly created Label tag.
+   */
+  function createLabel(tagName, name, parentTag, forId = undefined) {
+    const label = document.createElement(tagName);
+    if (forId) {
+      label.setAttribute('for', forId);
+    }
+    label.innerHTML = name;
+    parentTag.appendChild(label);
+    return label;
+  }
+
+  //  function renderLayer
+
+  /**
+   * Create checkboxes for given structure's layers and child structures.
+   * @param {Object} curStructure the structure generated from
+   renderSingleIBRStructure() that will be added to sidebar.
+   * @param {String} structureName the name of the structure being processed.
+   */
+  function drawSingleStructureSidebar(curStructure, structureName) {
+    // Create checkbox for each layer
+    if (curStructure['layers'].size !== 0) {
+      for ( const [layerName, layer] of
+        Object.entries(curStructure['layers']) ) {
+        const checkBox = document.createElement('INPUT');
+        const div = document.createElement('DIV');
+        checkBox.setAttribute('type', 'checkbox');
+        checkBox.setAttribute('id', structureName + '_' + layerName);
+        div.appendChild(checkBox);
+        createLabel('label', layerName, div, structureName + '_' +
+        layerName);
+        document.getElementById(structureName).appendChild(div);
+        checkBox.addEventListener('change', function() {
+          if (checkBox.checked) {
+            for ( const line of layer ) {
+              line.visible = true;
+            }
+          } else {
+            for ( const line of layer ) {
+              line.visible = false;
+            }
+          }
+        });
+      }
+    }
+
+    // Create label for each child structure
+    for ( let structureIndex = 0; structureIndex <
+    curStructure['structures'].length; structureIndex++ ) {
+      const li = document.createElement('li');
+      document.getElementById(structureName).appendChild(li);
+      const label = createLabel('span',
+          curStructure['structures'][structureIndex].name, li);
```

Comment:
this will break if we adjust the proto structure. we should distinguish and create a separate method for data parsing and pass in the output here

Reply:
curStructure is parsed from the proto object in function renderSingleIBRStructure(), curStructure['structures'] is a list of structure objects. I agree getter functions are more reliable in this case. Will add them in the next commit.

Reply:
sounds good :)